building/devices/zynqmp.rst

```diff
@@ -1,150 +1,102 @@
++------------+-------------------------------+
+| ZCU106     | ``PLATFORM=zynqmp-zcu106``    |
++------------+-------------------------------+
+| Ultra96    | ``PLATFORM=zynqmp-ultra96``   |
++------------+-------------------------------+
 
-.. code-block:: bash
-	
-	$ tee-supplicant -d
-	$ xtest
-	
-You can close QEMU session at any time by typing ``Ctrl-A+C`` and entering the 
-``quit`` command.
-
-Configuring and building for other ZynqMP boards
-*************************************************
-To use this makefile with other supported boards, you have to download the 
-corresponding BSP and add option ``PLATFORM`` to each make command.
+An example of fetch and build commands is:
 
 .. code-block:: bash
-	 
-	$ make -f zynqmp.mk PLATFORM=zcu106
-	$ make -f zynqmp.mk PLATFORM=zcu106 qemu
-
-Hereafter the list of available ``PLATFORM``:
-
-	* ``zcu102``
-	* ``zcu104``
-	* ``zcu106``
-	* ``ultra96-reva``
-
-.. warning::
-
-	On Ultra96 board, UART is not directly available. You have to connect
-	through WIFI Access Point using the procedure detailed here 
-	`Getting started`_.
-
-SD card creation
-****************
-After completion of building process, you can create a bootable SD card. Here,
-we consider that SD card corresponds to ``/dev/sdb``. We will use ``gparted``
-and ``e2image`` tools.
 
-Using ``gparted`` or any other partition manager tool create two partitions on 
-the card:
+  $ repo init -u https://github.com/OP-TEE/manifest.git -m zynqmp.xml
+  $ repo sync
+  $ cd build
+  $ make toolchains
+  $ make PLATFORM=zynqmp-zcu102 all
 
-	* 1GB FAT32 bootable partition (``/dev/sdb1`` hereafter).
-	* EXT4 partition on the remaining memory space (``/dev/sdb2``
-	  hereafter).
+After completion of the buildling process, two new files will be generated
+within the ``zynqmp/`` folder, ``BOOT.bin`` and ``<platform-name>.ub``. The
+first one is the boot image composed of the FSBL, PMU Firmware, ARM Trusted
+Firmware, OP-TEE and U-Boot. The second one is a FIT image containing the Linux
+kernel, the device-tree blob and the initramfs root file system.
 
-Once SD card is partitioned, use the following commands:
-
-.. code-block:: bash
-	 
-	$ cp /path/to/project/images/linux/BOOT.BIN /dev/sdb1
-	$ cp /path/to/project/images/linux/image.ub /dev/sdb1
-	$ sudo e2image -rap /path/to/project/images/linux/rootfs.ext4 /dev/sdb2
+.. note::
+	If the firmware image is not provided to the build script the boot image
+	will not be generated.
 
-Now you can use the newly created SD card to boot your board.
+Petalinux build instructions
+*****************************************
+OP-TEE build can be additionally integrated within Xilinx Petalinux tool for
+Embedded Linux development. As Petalinux is built in top of Yocto, the
+integration is performed through adding some exisiting recipes and few
+customizations. Use the previous build `Makefile
+<https://github.com/OP-TEE/build/blob/master/zynqmp.mk>`_ based on Petalinux
```

Comment:
If I correctly understand the description, I think the right reference should be https://github.com/OP-TEE/build/blob/3.14.0/zynqmp.mk, before _zynqmp.mk_ moved from petalinux to OP-TEE common build.

Reply:
Right, just fixed the reference